Flaky test: tests/chainerx_tests/unit_tests/routines_tests/test_loss.py::test_SoftmaxCrossEntropy

https://travis-ci.org/chainer/chainer/jobs/601719725#L4484

Occured in #8295.

```
[2019-10-23 10:13:00] ___________________ test_SoftmaxCrossEntropy_param_3_{t_dtype='int16', x_dtype='float16'}[native:0] ____________________
[2019-10-23 10:13:00] 
[2019-10-23 10:13:00] device = native:0, args = (), kwargs = {}
[2019-10-23 10:13:00] backend_config = <BackendConfig use_chainerx=True chainerx_device='native:0' use_cuda=False cuda_device=None use_cudnn='never' cudnn_deterministic=False autotune=False cudnn_fast_batch_normalization=False use_ideep='never'>
[2019-10-23 10:13:00] obj = <chainer.testing._bundle.TestSoftmaxCrossEntropy_param_3_{t_dtype='int16', x_dtype='float16'} object at 0x7f102941a7f0>

...

[2019-10-23 10:13:00] E           chainer.testing.function_link.FunctionTestError: double backward is not implemented correctly
[2019-10-23 10:13:00] E           
[2019-10-23 10:13:00] E           (caused by)
[2019-10-23 10:13:00] E           AssertionError: check_double_backward failed (eps=0.001 atol=0.0001 rtol=0.001)
[2019-10-23 10:13:00] E           input[0]:
[2019-10-23 10:13:00] E           array([[-0.78076172, -1.33984375],
[2019-10-23 10:13:00] E                  [-1.2578125 , -0.56689453]], shape=(2, 2), dtype=float16, device='native:0')
[2019-10-23 10:13:00] E           grad_output[0]:
[2019-10-23 10:13:00] E           array([0.04290771, 0.39892578], shape=(2,), dtype=float16, device='native:0')
[2019-10-23 10:13:00] E           grad_grad_input[0]:
[2019-10-23 10:13:00] E           array([[0.71972656, 0.72851562],
[2019-10-23 10:13:00] E                  [0.38500977, 0.64404297]], shape=(2, 2), dtype=float16, device='native:0')
[2019-10-23 10:13:00] E           
[2019-10-23 10:13:00] E           check_backward failed (eps=0.001 atol=0.0001 rtol=0.001)
[2019-10-23 10:13:00] E           inputs[0]:
[2019-10-23 10:13:00] E           array([[-0.78076172, -1.33984375],
[2019-10-23 10:13:00] E                  [-1.2578125 , -0.56689453]], shape=(2, 2), dtype=float16, device='native:0')
[2019-10-23 10:13:00] E           inputs[1]:
[2019-10-23 10:13:00] E           array([0.04290771, 0.39892578], shape=(2,), dtype=float16, device='native:0')
[2019-10-23 10:13:00] E           grad_outputs[0]:
[2019-10-23 10:13:00] E           array([[0.71972656, 0.72851562],
[2019-10-23 10:13:00] E                  [0.38500977, 0.64404297]], shape=(2, 2), dtype=float16, device='native:0')
[2019-10-23 10:13:00] E           directions[0]:
[2019-10-23 10:13:00] E           array([[0.19774412, 0.24381235],
[2019-10-23 10:13:00] E                  [0.34787308, -0.21185603]], shape=(2, 2), dtype=float64, device='native:0')
[2019-10-23 10:13:00] E           directions[1]:
[2019-10-23 10:13:00] E           array([0.7028306 , 0.49151123], shape=(2,), dtype=float64, device='native:0')
[2019-10-23 10:13:00] E           gradients (numeric):  array(0.06802642, shape=(), dtype=float64, device='native:0')
[2019-10-23 10:13:00] E           gradients (backward): array(0.06819657, shape=(), dtype=float64, device='native:0')
[2019-10-23 10:13:00] E           
[2019-10-23 10:13:00] E           x: numeric gradient, y: backward gradient
[2019-10-23 10:13:00] E           Not equal to tolerance rtol=0.001, atol=0.0001
[2019-10-23 10:13:00] E           
[2019-10-23 10:13:00] E           (mismatch 100.0%)
[2019-10-23 10:13:00] E            x: array(0.068026)
[2019-10-23 10:13:00] E            y: array(0.068197)
[2019-10-23 10:13:00] E           
[2019-10-23 10:13:00] E           assert_allclose failed: 
[2019-10-23 10:13:00] E             shape: () ()
[2019-10-23 10:13:00] E             dtype: float64 float64
[2019-10-23 10:13:00] E             i: (0,)
[2019-10-23 10:13:00] E             x[i]: 0.06802642484513102
[2019-10-23 10:13:00] E             y[i]: 0.06819657403670369
[2019-10-23 10:13:00] E             relative error[i]: 0.00249498151448332
[2019-10-23 10:13:00] E             absolute error[i]: 0.00017014919157266883
[2019-10-23 10:13:00] E             relative tolerance * |y[i]|: 6.819657403670369e-05
[2019-10-23 10:13:00] E             absolute tolerance: 0.0001
[2019-10-23 10:13:00] E             total tolerance: 0.0001681965740367037
[2019-10-23 10:13:00] E           x: 0.06802642
[2019-10-23 10:13:00] E           y: 0.06819657
[2019-10-23 10:13:00] E           
[2019-10-23 10:13:00] E           
[2019-10-23 10:13:00] E           (caused by)
[2019-10-23 10:13:00] E           AssertionError: check_backward failed (eps=0.001 atol=0.0001 rtol=0.001)
[2019-10-23 10:13:00] E           inputs[0]:
[2019-10-23 10:13:00] E           array([[-0.78076172, -1.33984375],
[2019-10-23 10:13:00] E                  [-1.2578125 , -0.56689453]], shape=(2, 2), dtype=float16, device='native:0')
[2019-10-23 10:13:00] E           inputs[1]:
[2019-10-23 10:13:00] E           array([0.04290771, 0.39892578], shape=(2,), dtype=float16, device='native:0')
[2019-10-23 10:13:00] E           grad_outputs[0]:
[2019-10-23 10:13:00] E           array([[0.71972656, 0.72851562],
[2019-10-23 10:13:00] E                  [0.38500977, 0.64404297]], shape=(2, 2), dtype=float16, device='native:0')
[2019-10-23 10:13:00] E           directions[0]:
[2019-10-23 10:13:00] E           array([[0.19774412, 0.24381235],
[2019-10-23 10:13:00] E                  [0.34787308, -0.21185603]], shape=(2, 2), dtype=float64, device='native:0')
[2019-10-23 10:13:00] E           directions[1]:
[2019-10-23 10:13:00] E           array([0.7028306 , 0.49151123], shape=(2,), dtype=float64, device='native:0')
[2019-10-23 10:13:00] E           gradients (numeric):  array(0.06802642, shape=(), dtype=float64, device='native:0')
[2019-10-23 10:13:00] E           gradients (backward): array(0.06819657, shape=(), dtype=float64, device='native:0')
[2019-10-23 10:13:00] E           
[2019-10-23 10:13:00] E           x: numeric gradient, y: backward gradient
[2019-10-23 10:13:00] E           Not equal to tolerance rtol=0.001, atol=0.0001
[2019-10-23 10:13:00] E           
[2019-10-23 10:13:00] E           (mismatch 100.0%)
[2019-10-23 10:13:00] E            x: array(0.068026)
[2019-10-23 10:13:00] E            y: array(0.068197)
[2019-10-23 10:13:00] E           
[2019-10-23 10:13:00] E           assert_allclose failed: 
[2019-10-23 10:13:00] E             shape: () ()
[2019-10-23 10:13:00] E             dtype: float64 float64
[2019-10-23 10:13:00] E             i: (0,)
[2019-10-23 10:13:00] E             x[i]: 0.06802642484513102
[2019-10-23 10:13:00] E             y[i]: 0.06819657403670369
[2019-10-23 10:13:00] E             relative error[i]: 0.00249498151448332
[2019-10-23 10:13:00] E             absolute error[i]: 0.00017014919157266883
[2019-10-23 10:13:00] E             relative tolerance * |y[i]|: 6.819657403670369e-05
[2019-10-23 10:13:00] E             absolute tolerance: 0.0001
[2019-10-23 10:13:00] E             total tolerance: 0.0001681965740367037
[2019-10-23 10:13:00] E           x: 0.06802642
[2019-10-23 10:13:00] E           y: 0.06819657
[2019-10-23 10:13:00] 
[2019-10-23 10:13:00] ../../../virtualenv/python3.5.6/lib/python3.5/site-packages/chainer/gradient_check.py:536: FunctionTestError
```

The double backward test also failed with `cuda` device.
https://jenkins.preferred.jp/job/chainer/job/chainer_pr/2349/TEST=CHAINERX_chainerx-py3,label=mn1-p100/console

<details>

```
16:17:45 _ test_SoftmaxCrossEntropy_param_9_{t_dtype='int64', x_dtype='float16'}[cuda:0] _
16:17:45 
16:17:45 device = cuda:0, args = (), kwargs = {}
16:17:45 backend_config = <BackendConfig use_chainerx=True chainerx_device='cuda:0' use_cuda=False cuda_device=None use_cudnn='never' cudnn_deterministic=False autotune=False cudnn_fast_batch_normalization=False use_ideep='never'>
16:17:45 obj = <chainer.testing._bundle.TestSoftmaxCrossEntropy_param_9_{t_dtype='int64', x_dtype='float16'} object at 0x7f4434f7e208>

...

16:17:45 E           chainer.testing.function_link.FunctionTestError: double backward is not implemented correctly
16:17:45 E           
16:17:45 E           (caused by)
16:17:45 E           AssertionError: check_double_backward failed (eps=0.001 atol=0.0001 rtol=0.001)
16:17:45 E           input[0]:
16:17:45 E           array([[0.59863281, -0.68798828],
16:17:45 E                  [-0.8149414 , -0.83398438]], shape=(2, 2), dtype=float16, device='cuda:0')
16:17:45 E           grad_output[0]:
16:17:45 E           array([-0.87402344, -0.2265625 ], shape=(2,), dtype=float16, device='cuda:0')
16:17:45 E           grad_grad_input[0]:
16:17:45 E           array([[-0.70703125, -0.89404297],
16:17:45 E                  [-0.65039062, -0.75146484]], shape=(2, 2), dtype=float16, device='cuda:0')
16:17:45 E           
16:17:45 E           check_backward failed (eps=0.001 atol=0.0001 rtol=0.001)
16:17:45 E           inputs[0]:
16:17:45 E           array([[0.59863281, -0.68798828],
16:17:45 E                  [-0.8149414 , -0.83398438]], shape=(2, 2), dtype=float16, device='cuda:0')
16:17:45 E           inputs[1]:
16:17:45 E           array([-0.87402344, -0.2265625 ], shape=(2,), dtype=float16, device='cuda:0')
16:17:45 E           grad_outputs[0]:
16:17:45 E           array([[-0.70703125, -0.89404297],
16:17:45 E                  [-0.65039062, -0.75146484]], shape=(2, 2), dtype=float16, device='cuda:0')
16:17:45 E           directions[0]:
16:17:45 E           array([[-0.26108988, -0.29465238],
16:17:45 E                  [0.49377820, -0.18927506]], shape=(2, 2), dtype=float64, device='cuda:0')
16:17:45 E           directions[1]:
16:17:45 E           array([-0.13504375, 0.73968458], shape=(2,), dtype=float64, device='cuda:0')
16:17:45 E           gradients (numeric):  array(0.03836282, shape=(), dtype=float64, device='cuda:0')
16:17:45 E           gradients (backward): array(0.03855777, shape=(), dtype=float64, device='cuda:0')
16:17:45 E           
16:17:45 E           x: numeric gradient, y: backward gradient
16:17:45 E           Not equal to tolerance rtol=0.001, atol=0.0001
16:17:45 E           
16:17:45 E           Mismatch: 100%
16:17:45 E           Max absolute difference: 0.00019495
16:17:45 E           Max relative difference: 0.00505601
16:17:45 E            x: array(0.038363)
16:17:45 E            y: array(0.038558)
16:17:45 E           
16:17:45 E           assert_allclose failed: 
16:17:45 E             shape: () ()
16:17:45 E             dtype: float64 float64
16:17:45 E             i: (0,)
16:17:45 E             x[i]: 0.038362820087562484
16:17:45 E             y[i]: 0.03855776859601062
16:17:45 E             relative error[i]: 0.005056011163164366
16:17:45 E             absolute error[i]: 0.0001949485084481381
16:17:45 E             relative tolerance * |y[i]|: 3.8557768596010625e-05
16:17:45 E             absolute tolerance: 0.0001
16:17:45 E             total tolerance: 0.00013855776859601063
16:17:45 E           x: 0.03836282
16:17:45 E           y: 0.03855777
16:17:45 E           
16:17:45 E           
16:17:45 E           (caused by)
16:17:45 E           AssertionError: check_backward failed (eps=0.001 atol=0.0001 rtol=0.001)
16:17:45 E           inputs[0]:
16:17:45 E           array([[0.59863281, -0.68798828],
16:17:45 E                  [-0.8149414 , -0.83398438]], shape=(2, 2), dtype=float16, device='cuda:0')
16:17:45 E           inputs[1]:
16:17:45 E           array([-0.87402344, -0.2265625 ], shape=(2,), dtype=float16, device='cuda:0')
16:17:45 E           grad_outputs[0]:
16:17:45 E           array([[-0.70703125, -0.89404297],
16:17:45 E                  [-0.65039062, -0.75146484]], shape=(2, 2), dtype=float16, device='cuda:0')
16:17:45 E           directions[0]:
16:17:45 E           array([[-0.26108988, -0.29465238],
16:17:45 E                  [0.49377820, -0.18927506]], shape=(2, 2), dtype=float64, device='cuda:0')
16:17:45 E           directions[1]:
16:17:45 E           array([-0.13504375, 0.73968458], shape=(2,), dtype=float64, device='cuda:0')
16:17:45 E           gradients (numeric):  array(0.03836282, shape=(), dtype=float64, device='cuda:0')
16:17:45 E           gradients (backward): array(0.03855777, shape=(), dtype=float64, device='cuda:0')
16:17:45 E           
16:17:45 E           x: numeric gradient, y: backward gradient
16:17:45 E           Not equal to tolerance rtol=0.001, atol=0.0001
16:17:45 E           
16:17:45 E           Mismatch: 100%
16:17:45 E           Max absolute difference: 0.00019495
16:17:45 E           Max relative difference: 0.00505601
16:17:45 E            x: array(0.038363)
16:17:45 E            y: array(0.038558)
16:17:45 E           
16:17:45 E           assert_allclose failed: 
16:17:45 E             shape: () ()
16:17:45 E             dtype: float64 float64
16:17:45 E             i: (0,)
16:17:45 E             x[i]: 0.038362820087562484
16:17:45 E             y[i]: 0.03855776859601062
16:17:45 E             relative error[i]: 0.005056011163164366
16:17:45 E             absolute error[i]: 0.0001949485084481381
16:17:45 E             relative tolerance * |y[i]|: 3.8557768596010625e-05
16:17:45 E             absolute tolerance: 0.0001
16:17:45 E             total tolerance: 0.00013855776859601063
16:17:45 E           x: 0.03836282
16:17:45 E           y: 0.03855777
16:17:45 
16:17:45 atol       = 0.0001
16:17:45 d          = array([-0.13504375, 0.73968458], shape=(2,), dtype=float64, device='cuda:0')
16:17:45 directions = (array([[-0.26108988, -0.29465238],
16:17:45        [0.49377820, -0.18927506]], shape=(2, 2), dtype=float64, device='cuda:0'), array([-0.13504375, 0.73968458], shape=(2,), dtype=float64, device='cuda:0'))
16:17:45 eps        = 0.001
16:17:45 f          = <_io.StringIO object at 0x7f44ab03a948>
16:17:45 gx_backward = array(0.03855777, shape=(), dtype=float64, device='cuda:0')
16:17:45 gx_numeric = array(0.03836282, shape=(), dtype=float64, device='cuda:0')
16:17:45 gy         = array([[-0.70703125, -0.89404297],
16:17:45        [-0.65039062, -0.75146484]], shape=(2, 2), dtype=float16, device='cuda:0')
16:17:45 gys        = (array([[-0.70703125, -0.89404297],
16:17:45        [-0.65039062, -0.75146484]], shape=(2, 2), dtype=float16, device='cuda:0'),)
16:17:45 i          = 1
16:17:45 rtol       = 0.001
16:17:45 self       = <chainer.gradient_check._CheckBackward object at 0x7f4434f72748>
16:17:45 x          = array([-0.87402344, -0.2265625 ], shape=(2,), dtype=float16, device='cuda:0')
16:17:45 xs         = (array([[0.59863281, -0.68798828],
16:17:45        [-0.8149414 , -0.83398438]], shape=(2, 2), dtype=float16, device='cuda:0'), array([-0.87402344, -0.2265625 ], shape=(2,), dtype=float16, device='cuda:0'))
16:17:45 
16:17:45 /workspace/conda/envs/testenv/lib/python3.6/site-packages/chainer/gradient_check.py:536: FunctionTestError
```

--- a/tests/chainerx_tests/unit_tests/routines_tests/test_loss.py
+++ b/tests/chainerx_tests/unit_tests/routines_tests/test_loss.py
@@ -189,8 +189,10 @@ class TestSoftmaxCrossEntropy(op_utils.ChainerOpTest):
         self.t = t.astype(self.t_dtype)
 
         if self.x_dtype == 'float16':
-            self.check_forward_options.update({'rtol': 5e-3, 'atol': 5e-4})
-            self.check_backward_options.update({'rtol': 5e-3, 'atol': 5e-4})
+            self.check_forward_options.update({'rtol': 5e-3, 'atol': 5e-3})
+            self.check_backward_options.update({'rtol': 1e-2, 'atol': 5e-3})
+            self.check_double_backward_options.update(
+                {'rtol': 1e-2, 'atol': 3e-1})
 
     def generate_inputs(self):
         x = numpy.random.normal(loc=0, scale=1.0, size=self.shape)

